libs/librepcb/projecteditor/boardeditor/boardeditor.cpp

```diff
@@ -861,17 +862,36 @@ QStringList BoardEditor::getSearchToolBarCompleterList() noexcept {
 
 void BoardEditor::goToDevice(const QString& name) noexcept {
   if (Board* board = getActiveBoard()) {
+    if (mLastSearch != name) {
+      mLastSearch = name;
+      mNextDeviceIndex = 0;
+    }
+    QList<BI_Device*> deviceCandidates = {};
     foreach (BI_Device* device, board->getDeviceInstances()) {
-      if (device->getComponentInstance().getName()->toLower() ==
-          name.toLower()) {
-        board->clearSelection();
-        device->setSelected(true);
-        QRectF rect   = device->getFootprint().getBoundingRect();
-        qreal  margin = Length(1000000).toPx();
-        rect.adjust(-margin, -margin, margin, margin);
-        mGraphicsView->zoomToRect(rect);
-        return;
+      if (device->getComponentInstance().getName()
+          ->startsWith(name, Qt::CaseInsensitive)) {
+        deviceCandidates.append(device);
+      }
+    }
+    std::sort(deviceCandidates.begin(), deviceCandidates.end(),
+              [](BI_Device* A, BI_Device* B){
+      return A->getComponentInstance().getName()
+          < B->getComponentInstance().getName();
+    });
+    if (deviceCandidates.count()) {
+      if (mNextDeviceIndex >= deviceCandidates.count()) {
+        mNextDeviceIndex = 0;
       }
+      BI_Device* device = deviceCandidates[mNextDeviceIndex++];
+      mNextDeviceIndex = mNextDeviceIndex % deviceCandidates.count();
+      board->clearSelection();
+      device->setSelected(true);
+      QRectF rect = device->getFootprint().getBoundingRect();
+      //Zoom to a rectangle relative to the maximum device dimension. The device
+      // is 1/4th of the screen
```

Comment:
Missing space :see_no_evil: 

```suggestion
      // Zoom to a rectangle relative to the maximum device dimension. The
      // device is 1/4th of the screen.
```